From 927051d13dd5f069c15efbb2c68d9e1145d83668 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 2 Feb 2017 01:46:12 +0000
Subject: [PATCH 4/4] metadata: Do not override fallback timestamp when release
 is inconsistent

This prevents the dates changing when solbuild passes a timestamp for
a history event that is yet to happen.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/metadata.py | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/ypkg2/metadata.py b/ypkg2/metadata.py
index f39e511..acdb9b2 100644
--- a/ypkg2/metadata.py
+++ b/ypkg2/metadata.py
@@ -75,17 +75,16 @@ def initialize_timestamp(spec):
     global history_timestamp
     global fallback_timestamp
 
+    dt = datetime.datetime.utcnow()
+    s = dt.strftime("%Y-%m-%d")
+    fallback_timestamp = utc_date_for_date_only(s)
+
     # If it was already set, just skip it.
     if history_timestamp is not None:
         dt = datetime.datetime.fromtimestamp(history_timestamp)
         history_date = dt.strftime("%Y-%m-%d")
-        fallback_timestamp = history_timestamp
         return
 
-    dt = datetime.datetime.utcnow()
-    s = dt.strftime("%Y-%m-%d")
-    fallback_timestamp = utc_date_for_date_only(s)
-
     if spec.history:
         up = spec.history.history[0]
         history_timestamp = utc_date_for_date_only(up.date)
-- 
2.11.0

